### PR TITLE
proc: avoid setting duplicate breakpoints when several goroutines wai…

### LINKED ISSUE #258: next failed, "Breakpoing exists at /go/src/github.com/docker/docker/vendor/src/github.com/godbus/dbus/conn.go:370"
Environment:

CentOS 7 (3.19.3)

$ go version
go version go1.4.2 linux/amd64

root@bao-dev:~/.../github.com/derekparker/delve
$ docker version
Client:
 Version:      1.8.1
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   fdd9957-dirty
 Built:        Thu Sep 17 15:51:19 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.0-dev
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   bba762b-dirty
 Built:        Sun Sep 27 01:04:09 UTC 2015
 OS/Arch:      linux/amd64
root@bao-dev:~/.../github.com/derekparker/delve

start docker with `docker daemon`

$ ps -ef | grep docker
root      8895  8894  0 02:02 ?        00:00:00 /usr/bin/cscope -dl -f /root/git/go/src/github.com/docker/docker/cscope.out
root     13500  2138  1 02:44 pts/1    00:00:00 ./bundles/1.9.0-dev/binary/docker daemon
root     13531 21784  0 02:44 pts/72   00:00:00 grep --color=auto docker
root@bao-dev:~/.../github.com/docker/docker
$ dlv attach 13500
Type 'help' for list of commands.
(dlv) b Containers
Breakpoint 1 set at 0x5a4841 for github.com/docker/docker/daemon.(*Daemon).Containers /go/src/github.com/docker/docker/daemon/list.go:31
(dlv) c
> github.com/docker/docker/daemon.(*Daemon).Containers() /go/src/github.com/docker/docker/daemon/list.go:31

   26:          Filters string
   27:  }
   28:
   29:  func (daemon *Daemon) Containers(config *ContainersConfig) ([]*types.Container, error) {
   30:          var (
=> 31:                  foundBefore bool
   32:                  displayed   int
   33:                  all         = config.All
   34:                  n           = config.Limit
   35:                  psFilters   filters.Args
   36:                  filtExited  []int

(dlv) bp
Breakpoint 1 at 0x5a4841 /go/src/github.com/docker/docker/daemon/list.go:31
(dlv) n
Command failed: Breakpoint exists at /go/src/github.com/docker/docker/vendor/src/github.com/godbus/dbus/conn.go:370 at ce2e4d
(dlv) next
Command failed: Breakpoint exists at /go/src/github.com/docker/docker/vendor/src/github.com/godbus/dbus/conn.go:370 at ce2e4d
(dlv)

Set a break point at `docker ps` (Containers()), continue, do `docker ps` from command line.  dlv caught the break point, but unable to step through the code.

### PR DESCRIPTION
**Pull request by kostya-sh**
…t on the same channel

I encountered the error message described in the issue while debugging
InfluxDB. I can confirm that after this fix the error disapeared.

Fixes #258

### CODE REVIEW

File: proc/proc.go [hunk 1]
```diff
@@ -343,9 +345,15 @@ func (dbp *Process) setChanRecvBreakpoints() (int, error) {
 				}
 				return 0, err
 			}
+			if _, ok := setBPs[ret]; ok {
+				// Do not set duplicate breakpoint in case if multiple
+				// goroutines waiting on the same channel
+				continue
+			}
 			if _, err = dbp.SetTempBreakpoint(ret); err != nil {
 				return 0, err
```
Comment (derekparker) [inline proc/proc.go]: Instead off keeping track of what we've set, let's just check for `err.(BreakpointExistsError)` and ignore it.

Comment (kostya-sh) [inline proc/proc.go]: Done

**Comment by derekparker** [comment]
LGTM, I'm going to squash the commits and then merge.

**Comment by derekparker** [comment]
Merged.

**Pull request closed without merging**